collection-scripts/monitoring_common.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+# safeguards
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# global readonly constants
+declare -r METRICS_PATH="../must-gather/monitoring/metrics"
+
+get_metrics_min_time() {
+  # default to the last 2h as Prometheus servs that from memory.
+  promtool_min_time=$(date --date='2 hours ago' +%s%3N)
+  # takes MUST_GATHER_SINCE into account.
+  # TODO: update description in "oc"
+  if [ -n "${MUST_GATHER_SINCE:-}" ]; then
+    promtool_min_time=$(date --date=${MUST_GATHER_SINCE} +%s%3N)
+  fi
+}
+
+metrics_get() {
+  mkdir -p "${METRICS_PATH}"
+
+  readarray -t READY_PROM_PODS < <(
+    oc get pods -n openshift-monitoring  -l prometheus=k8s --field-selector=status.phase==Running \
+      --no-headers -o custom-columns=":metadata.name"
+  )
+  prometheus_pod=${READY_PROM_PODS[0]}
+  echo "INFO: Getting metrics from ${prometheus_pod}"
+
+  oc exec ${prometheus_pod} \
+    -c prometheus \
+    -n openshift-monitoring \
+    -- promtool tsdb dump-openmetrics /prometheus "$@" \
+       > "$METRICS_PATH/metrics.txt" \
+       2> "$METRICS_PATH/metrics.stderr"
+}
```

Comment:
tried with
`$ oc -n openshift-monitoring exec prometheus-k8s-0  -c prometheus -- promtool tsdb dump-openmetrics /prometheus/ > ~/metrics.txt 2> ~/metrics.stderr`
metrics.txt size is 150M, it will be not easy to view the file with txt editor if the size is much more larger
```
$ du -h metrics.txt 
150M	metrics.txt
```

Reply:
Yes the dump can be heavy as mentioned in the command "help" below.
Besides viewing it in an IDE, the results can be processed in a terminal or restored into a new/local prometheus instance.

Reply:
Should the output file perhaps be called `metrics.openmetrics` or so?
We could also add a small script that takes a tsdb directory argument and loads the metrics? Maybe even show some helpful metadata, like the time period that the new metrics live in. If promtool doesn't do that already that is.
Nothing that should block this PR imo though.

Reply:
> Should the output file perhaps be called metrics.openmetrics or so?

Good call, done.

> We could also add a small script that takes a tsdb directory argument and loads the metrics? Maybe even show some helpful metadata, like the time period that the new metrics live in. If promtool doesn't do that already that is.

yes, once we identify the other operations that users need/want to run besides `promtool tsdb create-blocks-from openmetrics`, we can propose a wrapper around that (if nobody had done it before :)), for now, I think an empty 2.x Prometheus instance + `promtool tsdb create-blocks-from openmetrics` should be sufficient. (I'll provide some doc about that)

yes, `promtool tsdb create-blocks-from openmetrics` logs some info about the resulted blocks.